cloudevents/working-drafts/dds-cloudevent.xml

```diff
@@ -0,0 +1,105 @@
+            <member name="content-type" type="string" stringMaxLength="255" optional="true"/>
+          </struct>
+	  
+	  <enum name="AttributeType">
+            <enumerator name="BOOL"/>
+            <enumerator name="INT32"/>
+            <enumerator name="STRING"/>
+            <enumerator name="BYTES"/>
+            <enumerator name="URI"/>
+            <enumerator name="URI_REF"/>
+            <enumerator name="TIMESTAMP"/>
+          </enum>
+	  
+          <union name="AttributeValue" extensibility="final">
+            <discriminator type="nonBasic" nonBasicTypeName="io::cloudevents::AttributeType"/>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::BOOL)"/>
+              <member name="ce_boolean" type="boolean"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::INT32)"/>
+              <member name="ce_integer" type="int32"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::STRING)"/>
+              <member name="ce_string" type="string" stringMaxLength="255"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::BYTES)"/>
+              <member name="ce_bytes" type="byte" sequenceMaxLength="100"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::URI)"/>
+              <member name="ce_uri" type="nonBasic" nonBasicTypeName="io::cloudevents::URI"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::URI_REF)"/>
+              <member name="ce_uri_reference" type="nonBasic" nonBasicTypeName="io::cloudevents::URI_Ref"/>
+            </case>
+	  </union>
+
+	   <struct name="DDS-CE-Attribute">
+            <member name="key" type="string" stringMaxLength="255"/>
+            <member name="value" type="nonBasic" nonBasicTypeName="io::cloudevents::AttributeValue"/>
+          </struct>
+          <typedef name="DDS-CE-Attributes" type="nonBasic" nonBasicTypeName="io::cloudevents::DDS-CE-Attribute" sequenceMaxLength="100"/>
+	    
+          <enum name="DataKind">
+            <enumerator name="BINARY"/>
+            <enumerator name="TEXT"/>
+          </enum>
+	  
+          <union name="Data" extensibility="final">
+            <discriminator type="nonBasic" nonBasicTypeName="io::cloudevents::DataKind"/>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::BINARY)"/>
+              <member name="binary_data" type="byte" sequenceMaxLength="100"/>
+            </case>
+            <case>
+              <caseDiscriminator value="(io::cloudevents::TEXT)"/>
+              <member name="text_data" type="string" stringMaxLength="255"/>
+            </case>
+          </union>
+
+          <struct name="Event" extensibility="mutable">
+            <member name="headers" type="nonBasic" nonBasicTypeName="io::cloudevents::Headers"/>
```

Comment:
What does this represent ?

Reply:
The DDS protocol doesn't have any mechanism for sending headers as some other protocols do so we have to create a custom header structure within the DDS message for the users to add their headers.

Reply:
if @JemDay is correct and "headers" are used for extensions, then we may need to rethink this because CE extensions are meant to be serialized just like spec-defined attributes, not within a separate "bucket"